### PR TITLE
Added function find_buildernames to help with issue 168

### PR DESCRIPTION
**Pull request by adusca**

### CODE REVIEW

**Comment by vaibhavmagarwal** [comment]
Awesome @adusca . I think that "function find_buildernames" is a bit too specific and should not be in platforms.py, but we can move it into a separate script afterwards. And nice removal of extra trailing spaces.

File: mozci/platforms.py [hunk 1]
```diff
@@ -257,3 +257,30 @@ def build_talos_buildernames_for_repo(repo_name, pgo_only=False):
 
     retVal.sort()
     return retVal
+
+
+def find_buildernames(repo, test=None, platform=None):
```
Comment (armenzg) [inline mozci/platforms.py:262]: I'm not suggesting to change the name but "filtered_builders" could also be another name for it if "find" does not work out.

File: mozci/platforms.py [hunk 2]
```diff
@@ -257,3 +257,30 @@ def build_talos_buildernames_for_repo(repo_name, pgo_only=False):
...
+    """
+    Return a list of buildernames matching the criteria.
+
+    1) if the developer provides test, repo and platform, return only the specific buildername
+    2) if the developer provides test and platform only, then return the test on all platforms
+    3) if the developer provides platform and repo, then return all the tests on that platform
+    """
+    if test is None and platform is None:
```
Comment (armenzg) [inline mozci/platforms.py:270]: You can have an assertion in here
assert true, "Message to be raised"

File: mozci/platforms.py [hunk 3]
```diff
@@ -257,3 +257,30 @@ def build_talos_buildernames_for_repo(repo_name, pgo_only=False):
...
+    matching = []
+    buildernames = _filter_builders_matching(fetch_allthethings_data()['builders'].keys(), repo)
+
+    if test is not None:
+        buildernames = _filter_builders_matching(buildernames, test)
+
+    if platform is not None:
+        for buildername in buildernames:
+            if get_associated_platform_name(buildername) == platform:
+                if is_downstream(buildername):
```
Comment (armenzg) [inline mozci/platforms.py:282]: Would this work _only_ with downstream jobs? (aka test jobs)

**Comment by armenzg** [comment]
You're such a pro at writing tests and understanding business needs. Well done!

Please merge once the comments are addressed.

**Comment by adusca** [comment]
Thanks!

If I removed "if is_downstream(buildername):" we would get every test job and a build job when the test is not specified. I asked @vaibhavmagarwal and he believes having only test jobs fits #168 better, because the build job would be automatically triggered if no suitable one is found. What do you think?

**Comment by armenzg** [comment]
Sure!
As long as we're aware of what we're doing.
If we need to change in the future we can.

**Pull request merged**